WordPress/src/main/java/org/wordpress/android/ui/reader/usecases/ReaderFetchRelatedPostsUseCase.kt

```diff
@@ -0,0 +1,78 @@
+package org.wordpress.android.ui.reader.usecases
+
+import kotlinx.coroutines.suspendCancellableCoroutine
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.models.ReaderPost
+import org.wordpress.android.ui.reader.ReaderEvents.RelatedPostsUpdated
+import org.wordpress.android.ui.reader.actions.ReaderPostActionsWrapper
+import org.wordpress.android.ui.reader.models.ReaderSimplePostList
+import org.wordpress.android.ui.reader.usecases.ReaderFetchRelatedPostsUseCase.FetchRelatedPostsState.AlreadyRunning
+import org.wordpress.android.ui.reader.usecases.ReaderFetchRelatedPostsUseCase.FetchRelatedPostsState.Failed.NoNetwork
+import org.wordpress.android.ui.reader.usecases.ReaderFetchRelatedPostsUseCase.FetchRelatedPostsState.Failed.RequestFailed
+import org.wordpress.android.ui.reader.usecases.ReaderFetchRelatedPostsUseCase.FetchRelatedPostsState.Success
+import org.wordpress.android.util.NetworkUtilsWrapper
+import javax.inject.Inject
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.resume
+
+class ReaderFetchRelatedPostsUseCase @Inject constructor(
+    private val networkUtilsWrapper: NetworkUtilsWrapper,
+    private val readerPostActionsWrapper: ReaderPostActionsWrapper
+) {
+    private val continuations: MutableMap<RelatedPostsRequest, Continuation<FetchRelatedPostsState>?> = mutableMapOf()
+
+    suspend fun fetchRelatedPosts(sourcePost: ReaderPost): FetchRelatedPostsState {
+        val request = RelatedPostsRequest(postId = sourcePost.postId, blogId = sourcePost.blogId)
+
+        return when {
+            continuations[request] != null -> {
+                AlreadyRunning
+            }
+            !networkUtilsWrapper.isNetworkAvailable() -> {
+                NoNetwork
+            }
```

Comment:
Minor (🔍): Consider removing braces for the `AlreadyRunning` and `NoNetwork` when entries. This will save us a few lines and will make it a bit more readable, wdyt?

Reply:
In 16fb72f